src/event_data.cc

```diff
@@ -195,7 +204,7 @@ namespace grpc_labview
             }
             else
             {
-                _stream.Finish(grpc::Status::CANCELLED, this);
+                _stream.Finish(grpc::Status(grpc::StatusCode::UNIMPLEMENTED, ""), this);
```

Comment:
do we also want to set _status = CallStatus:Finish here

Reply:
Good catch. We should never get into this scenario since we have already had a check during the read phase itself but set the status anyway.